src/Indexer.php

```diff
@@ -169,4 +161,17 @@ private function refreshBulkRequestParams()
             $this->currentBulk->setRequestParam($key, $value);
         }
     }
+
+    private function updateDocumentData($document): void
+    {
+        if ($document instanceof Document) {
+            $context = $this->contextBuilder->buildContext(\get_class($document->getDto()));
+            $data = $this->serializer->serialize($document->getData(), 'json', $context);
+            $document->setData($data);
+        } elseif ($document instanceof ElasticaDocument) {
+            // @phpstan-ignore-next-line
```

Comment:
```suggestion
        } elseif ($document instanceof ElasticaDocument && \is_object($document->getData())) {
```

To keep support of string|array